src/strategies/crucible-erc20-single-lp-token-balance/index.ts

```diff
@@ -0,0 +1,194 @@
+    { blockTag }
+  );
+
+  const lpTokenTotalSupply = await call(
+    provider,
+    abi,
+    [options.lpTokenAddress, 'totalSupply', []],
+    { blockTag }
+  );
+
+  let tokenWeight;
+
+  if (token0Address === options.tokenAddress) {
+
+    tokenWeight = (parseFloat(formatUnits(lpTokenReserves._reserve0, tokenDecimals)) / parseFloat(formatUnits(lpTokenTotalSupply, lpDecimals))) * 2
+
+  } else if (token1Address === options.tokenAddress) {
+
+    tokenWeight = (parseFloat(formatUnits(lpTokenReserves._reserve1, tokenDecimals)) / parseFloat(formatUnits(lpTokenTotalSupply, lpDecimals))) * 2
+
+  } else {
+
+    tokenWeight = 0;
+
+  }
+
+  // get the number of crucibles owned by the wallet
+  // wallet_address => crucible_count
+
+  const callWalletToCrucibleCount = new Multicaller(network, provider, abi, {
+    blockTag
+  });
+  for (const walletAddress of addresses) {
+    callWalletToCrucibleCount.call(
+      walletAddress,
+      options.crucible_factory,
+      'balanceOf',
+      [walletAddress]
+    );
+  }
+  const walletToCrucibleCount: Record<string, BigNumber> =
+    await callWalletToCrucibleCount.execute();
+
+  // get the address of each crucible
+  // wallet_address : crucible_index => crucible_address
+
+  const callWalletToCrucibleAddresses = new Multicaller(
+    network,
+    provider,
+    abi,
+    {
+      blockTag
+    }
+  );
+  for (const [walletAddress, crucibleCount] of Object.entries(
+    walletToCrucibleCount
+  )) {
+    for (let index = 0; index < crucibleCount.toNumber(); index++) {
+      callWalletToCrucibleAddresses.call(
+        walletAddress.toString() + '-' + index.toString(),
+        options.crucible_factory,
+        'tokenOfOwnerByIndex',
+        [walletAddress, index]
+      );
+    }
+  }
+  const walletIDToCrucibleAddresses: Record<string, BigNumber> =
+    await callWalletToCrucibleAddresses.execute();
+
+  // get the lp token balance of each crucible
+  // crucible_address => lp_balance
+
+  const callCrucibleToLpBalance = new Multicaller(network, provider, abi, {
+    blockTag
+  });
+  for (const [walletID, crucibleAddress] of Object.entries(
+    walletIDToCrucibleAddresses
+  )) {
+    callCrucibleToLpBalance.call(walletID, options.lpTokenAddress, 'balanceOf', [
+      hexZeroPad(crucibleAddress.toHexString(), 20)
+    ]);
+  }
+  const walletIDToLpBalance: Record<string, BigNumber> =
+    await callCrucibleToLpBalance.execute();
+
+  // get the token balance of each crucible
+  // crucible_address => token_balance
+
+  const callCrucibleToTokenBalance = new Multicaller(network, provider, abi, {
+    blockTag
+  });
+  for (const [walletID, crucibleAddress] of Object.entries(
+    walletIDToCrucibleAddresses
+  )) {
+    callCrucibleToTokenBalance.call(walletID, options.tokenAddress, 'balanceOf', [
+      hexZeroPad(crucibleAddress.toHexString(), 20)
+    ]);
+  }
+  const walletIDToTokenBalance: Record<string, BigNumber> =
+    await callCrucibleToTokenBalance.execute();
```

Comment:
This strategy is crossing the limit of 5 requests, make sure to follow this checklist https://github.com/snapshot-labs/snapshot-strategies#checklist-for-adding-a-new-strategy